narwhals/_pandas_like/group_by.py

```diff
@@ -237,43 +250,72 @@ def agg_pandas(  # noqa: PLR0915
             new_names = [new_names[i] for i in index_map]
             result_simple_aggs.columns = new_names
 
+            result_aggs.append(result_simple_aggs)
+
         if nunique_aggs:
             result_nunique_aggs = grouped[list(nunique_aggs.values())].nunique(
                 dropna=False
             )
             result_nunique_aggs.columns = list(nunique_aggs.keys())
-        if simple_aggs and nunique_aggs:
-            if (
-                set(result_simple_aggs.columns)
-                .difference(keys)
-                .intersection(result_nunique_aggs.columns)
-            ):
+
+            result_aggs.append(result_nunique_aggs)
+
+        if std_aggs:
+            result_aggs.extend(
+                [
+                    rename(
+                        grouped[list(output_to_root_name_mapping.values())].std(
+                            ddof=ddof
+                        ),
+                        # Invert the dict to have root_name: output_name
+                        # TODO(FBruzzesi): Account for duplicates
+                        columns={v: k for k, v in output_to_root_name_mapping.items()},
```

Comment:
🤔 in case someone does `.agg(nw.col('a').std(ddof=1).alias('b'), nw.col('a').std(ddof=1).alias('c'))`? hmm yeah i guess it's possible

btw, since cuDF have no introduced `cudf.NamedAgg`, we could use that if it makes all this logic a bit easier

Reply:
Yes correct, similarly to simple aggs.
To avoid an explosion in the number of lists to keep track of, maybe there is a better data structure. I will think about it a bit, yet aside that specific case, this PR should be ready